streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -373,14 +352,14 @@ public static Map<ProcessId, KafkaStreamsAssignment> optimizeRackAwareStandbyTas
             taskMoved = false;
             round++;
             for (int i = 0; i < kafkaStreamsAssignments.size(); i++) {
-                final UUID clientId1 = clientIds.get(i);
-                final KafkaStreamsAssignment clientState1 = kafkaStreamsAssignments.get(new ProcessId(clientId1));
+                final ProcessId clientId1 = clientIds.get(i);
+                final KafkaStreamsAssignment clientState1 = kafkaStreamsAssignments.get(clientId1);
```

Comment:
nit: kind of weird to call this "clientStateN" since "state" sounds more like "KafkaStreamsState" -- can you rename to `clientAssignment`?